File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int a = 0;
        int b = 0;
        int row = 0;
        int column = 0;
        for (int i = 1; i <= yellow; i++) {
            if(yellow % i == 0) {
                a = i;
                b = yellow / a;
                System.out.println("a = " + a);
                System.out.println("b = " + b);
            }
            if (2*(a+b+2) == brown) {
                row = b;
                column = a;
                System.out.println("row = " + row);
                System.out.println("column = " + column);
                break;
            }
        }

        int[] answer = {row + 2, column + 2};
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for i in range(1, yellow + 1):
        if yellow % i == 0:
            a = i
            b = yellow / i
            if (a *2 + b * 2 + 4) == brown:
                return [b+2, a+2]